packages/itwinui-react/src/core/TransferList/TransferList.tsx

```diff
@@ -0,0 +1,282 @@
+    }
+  };
+
+  return (
+    <ListItem
+      ref={ref}
+      onClick={onClickEvents}
+      onKeyDown={onKeyDown}
+      active={active}
+      actionable={actionable}
+      tabIndex={-1}
+      role={'option'}
+      aria-disabled={disabled ? 'true' : undefined}
+      aria-selected={active ? 'true' : undefined}
+      disabled={disabled}
+      {...rest}
+    >
+      {children}
+    </ListItem>
+  );
+}) as PolymorphicForwardRefComponent<'li', TransferListItemOwnProps>;
+TransferListItem.displayName = 'TransferList.Item';
+
+// ----------------------------------------------------------------------------
+// TransferList.ListboxLabel component
+
+type TransferListListboxLabelOwnProps = {
+  /**
+   * Text label that is wrapped by `TransferList.ListboxLabel`
+   */
+  children?: string;
+};
+
+const TransferListListboxLabel = React.forwardRef((props, ref) => {
+  const { children, className, id, ...rest } = props;
+
+  const { labelId, setLabelId } = useSafeContext(TransferListContext);
+
+  React.useEffect(() => {
+    if (id && id !== labelId) {
+      setLabelId(id);
+    }
+  }, [id, labelId, setLabelId]);
+
+  return (
+    <Box
+      as='div'
+      className={cx('iui-transfer-list-listbox-label', className)}
+      id={labelId}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TransferListListboxLabelOwnProps>;
+TransferListListboxLabel.displayName = 'TransferList.ListboxLabel';
+
+// ----------------------------------------------------------------------------
+// TransferList.Toolbar component
+
+const TransferListToolbar = polymorphic('iui-transfer-list-toolbar', {
+  role: 'toolbar',
+});
+TransferListToolbar.displayName = 'TransferList.Toolbar';
+
+/**
+ * The TransferList component is used to display a list within a box
+ * @example
+ * <TransferList>
+ *   <TransferList.ListboxWrapper>
+ *     <TransferList.Listbox>
+ *       <TransferList.Item>Item 1</TransferList.Item>
+ *       <TransferList.Item>Item 2</TransferList.Item>
+ *       <TransferList.Item>Item 3</TransferList.Item>
+ *       <TransferList.Item>Item 4</TransferList.Item>
+ *       <TransferList.Item>Item 5</TransferList.Item>
+ *       <TransferList.Item>Item 6</TransferList.Item>
+ *     </TransferList.Listbox>
+ *   </TransferList.ListboxWrapper>
+ * </TransferList>
+ */
+export const TransferList = Object.assign(TransferListComponent, {
+  /**
+   * 	TransferList listbox wrapper subcomponent
+   */
+  ListboxWrapper: TransferListListboxWrapper,
+  /**
+   * 	TransferList listbox subcomponent
+   */
+  Listbox: TransferListListbox,
+  /**
+   * 	TransferList item subcomponent
+   */
+  Item: TransferListItem,
+  /**
+   * 	TransferList listbox label subcomponent
+   */
+  ListboxLabel: TransferListListboxLabel,
+  /**
+   * 	TransferList toolbar subcomponent
+   */
+  Toolbar: TransferListToolbar,
+});
+
+export const TransferListContext = React.createContext<
```

Comment:
Is it a valid use case for ListBox to have disabled state (not individual list items, but full box disabled)?

Reply:
Not that I can think of, but we can always revisit later if someone requests it.